### PR TITLE
Customise title of the Cancel button

### PR DESCRIPTION
**Pull request by ctarda**
When the media picker is presented using `WPNavigationMediaPickerViewController`, the picker is embedded in a navigation controller.

On the left hand side of that navigation controller there is a `Cancel` button, that dismisses the picker.

The button is a system button of type `UIBarButtonSystemItemCancel`.

This PR adds an optional method to `WPMediaPickerViewControllerDelegate` so that implementations of that delegate can provide a custom string for the button title. 

If that optional method is not implemented, the picker will default to the `UIBarButtonSystemItemCancel` button

### CODE REVIEW

**Comment by SergioEstevao** [comment]
Agree with @frosty , this need to be reflected on the Group/Album selector also.

**Comment by ctarda** [comment]
@frosty @SergioEstevao I just pushed another commit to address that

**Comment by SergioEstevao** [comment]
@ctarda is this value something that will change dynamically, has the state of picker changes? or will always be the same after set? 

If always the same maybe a better place for this will be in the `WPMediaPickerOptions` class? Instead of adding one more delegate method.

**Comment by frosty** [comment]
@SergioEstevao Hmm, or even just a property on the picker or navigation picker itself? `cancelButtonTitle`?

**Comment by ctarda** [comment]
I did consider making this another property in the `WPMediaPickerOptions` but all its properties where boolean toggles, so I didn't want to break the symmetry of that object's API.

I also did consider a property in the navigation picker itself. I went with a delegate method because this seems to be like the kind of customisation that can be delegated, in particular, if like now it's an optional method.

Now that I take another look at the API of the `WPNavigationMediaPickerViewController`, and the fact that there are already a few other properties that are strings, I am going to move it there. But I think this, as well as those other properties that are customisation are better fit for delegation. Give me a few minutes...

**Comment by ctarda** [comment]
@SergioEstevao @frosty shall we give it another go? 😜

File: Pod/Classes/WPNavigationMediaPickerViewController.m [hunk 1]
```diff
@@ -95,9 +95,7 @@ - (void)setupNavigationController
     UINavigationController *nav = [[UINavigationController alloc] initWithRootViewController: rootController];
     nav.delegate = self;
 
-    if (!self.showGroupSelector) {
```
[the remark this answers is not in the document]
Comment (SergioEstevao) [reply Pod/Classes/WPNavigationMediaPickerViewController.m]: I think the rational behind this is that if you are not showing the group selector you want to have a Cancel directly on the media picker screen. But If you are showing the group selector you want to have a Back button there?

[the remark this answers is not in the document]
Comment (SergioEstevao) [reply Pod/Classes/WPNavigationMediaPickerViewController.m]: Ok so I tested this and all is working ok, I think the other delegate method was being used when we showed the group selector has a pop up.

So all is ok and can be shipped.

**Pull request merged**